db_stress_tool/db_stress_common.h

```diff
@@ -327,16 +330,97 @@ extern inline std::string Key(int64_t val) {
   return big_endian_key;
 }
 
+// A struct for maintaining the parameters for generating variable length keys
+struct KeyGenContext {
+  // Max length of the key in 8-byte units. Each level is a int64 encoded in
+  // big endian format
+  unsigned int levels;
+  // Number of keys of each possible length in a given window
+  std::vector<uint64_t> weights;
+  // Size of window for varying the key length
+  uint64_t window;
+};
+extern KeyGenContext key_gen_ctx;
+
+// Generate a variable length key string from the given int64 val. The
```

Comment:
I think you should mention that order is preserved, though I find it odd that it takes int64_t and we coerce to uint64_t

Reply:
Yes, it is a bit strange. The call sites expect int64_t as max_key is int64_t. The inverse, ```GetIntVal()``` takes a uint64_t*. I think it should be uint64_t throughout.